imagetest/test_suites/image_validation/license_test.go

```diff
@@ -0,0 +1,192 @@
+	`Brocade Linux Fibre Channel HBA Firmware`,
+	`QLogic Linux Fibre Channel HBA Firmware`,
+	`Unlimited distribution and/or modification is allowed as long as this copyright notice remains intact.`,
+	`Permission is hereby granted to use.*this.*for any purpose`,
+	`are in the public domain`,
+	`is (available|distributed) under the terms of the GNU.*Public License`,
+	`(libudev|libgudev|udev) is licensed under the GNU (L|)GPL`,
+	`The Linux Console Tools are covered by the GPL`,
+	`Some portions of os-prober`,
+	`Netcat and the associated package is a product of Avian Research, and is freely available in full source form with no restrictions save an obligation to give credit where due.`,
+	`Permission is hereby granted, without written agreement and without licence or royalty fees, to use, copy, modify, and distribute this software`,
+	`Open Market permits you to use, copy, modify, distribute, and license this Software and the Documentation for any purpose, provided that existing copyright notices are retained in all copies and that this notice is included verbatim in any distributions. No written agreement, license, or royalty fee is required for any of the authorized uses.`,
+	`This software is made available under the terms of *either* of the licenses found in LICENSE.APACHE or LICENSE.BSD. Contributions to cryptography are made under the terms of *both* these licenses.`,
+}
+
+const (
+	copyright = "/usr/share/doc/*/copyright"
+	license   = "/usr/share/doc/*/LICENSE"
+)
+
+func isValidLicenseName(licenseCheck string) bool {
+	for _, name := range licenseNames {
+		// (?i) case insensitive
+		var regexString = fmt.Sprintf(`(?i)((?:(?:License|Copyright)\s*:\s*%[1]s)|(?:(?:covered )*under (?:the )?%[1]s)|(?:under (?:the terms of )*the %[1]s))`, name)
+		re := regexp.MustCompile(regexString)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicenseText(licenseCheck string) bool {
+	for _, licenseText := range licenses {
+		// (?i) case insensitive
+		re := regexp.MustCompile(`(?i)(` + licenseText + `)`)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func TestArePackagesLegal(t *testing.T) {
+	var filenames []string
+	if utils.IsTargetLinux(utils.RedHat) || utils.IsTargetLinux(utils.SUSE) {
+		filenames, _ = filepath.Glob(license)
+	} else if utils.IsTargetLinux(utils.Debian) {
+		filenames, _ = filepath.Glob(copyright)
+	} else {
+		t.Skip("can not run test on other os")
+	}
+	for _, filename := range filenames {
+		if !isPackageLegal(filename) {
+			t.Fatalf("The packages are not legal to use")
+		}
+	}
+}
+
+func isPackageLegal(filepath string) bool {
+	bytes, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		log.Printf("error read file")
+	}
+
+	var licenseCheck string = string(bytes)
+
+	// Remove comment
+	re := regexp.MustCompile(`(\*|#)*`)
```

Comment:
you probably want + (1-or-more) rather than * (0-or-more) operator
and this change can almost certainly be done with strings.ReplaceAll or regexp.Regexp.ReplaceAllString

Reply:
Agree 1-or-more is more reasonable.